server/pipeline/restart.go

```diff
@@ -85,6 +87,10 @@ func Restart(ctx context.Context, store store.Store, lastBuild *model.Build, use
 
 	newBuild, buildItems, err := createBuildItems(ctx, store, newBuild, user, repo, pipelineFiles, envs)
 	if err != nil {
+		var parseErr *yaml.PipelineParseError
+		if errors.As(err, &parseErr) {
```

Comment:
why not .Is ?

Reply:
I don't think you can check typed errors with fields with `Is` 🤔

Reply:
did not looked up this ... I'll check

Reply:
"... sets target to that error value and returns true" ...

we should not use As at all ...

Reply:
https://go.dev/play/p/zcJL_wTcC0q

it will always be true ...